commit bd3ccc49712dc44911bf0913ca49815f59747d6f
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Wed Aug 20 21:09:10 2014 +0200

    Add __pthread_key_create alias
    
    Some versions of std::thread use it to detect the presence of libpthread (don't
    ask why)
    
    * sysdeps/hurd/pt-key-create.c (pthread_key_create): Rename to
    __pthread_key_create.
    (pthread_key_create): Define as strong alias of __pthread_key_create.
    * Versions (libpthread): Add __pthread_key_create.

diff --git a/libpthread/Versions b/libpthread/Versions
index 41bff96..19ff26c 100644
--- a/libpthread/Versions
+++ b/libpthread/Versions
@@ -88,6 +88,7 @@ libpthread {
     pthread_join;
 
     pthread_key_create; pthread_key_delete;
+    __pthread_key_create;
 
     pthread_kill;
 
diff --git a/libpthread/sysdeps/hurd/pt-key-create.c b/libpthread/sysdeps/hurd/pt-key-create.c
index b3e0141..42c7c85 100644
--- a/libpthread/sysdeps/hurd/pt-key-create.c
+++ b/libpthread/sysdeps/hurd/pt-key-create.c
@@ -31,7 +31,7 @@ int __pthread_key_count;
 int __pthread_key_invalid_count;
 
 int
-pthread_key_create (pthread_key_t *key, void (*destructor) (void *))
+__pthread_key_create (pthread_key_t *key, void (*destructor) (void *))
 {
   /* Where to look for the next key slot.  */
   static int index;
@@ -107,3 +107,5 @@ pthread_key_create (pthread_key_t *key, void (*destructor) (void *))
   __pthread_mutex_unlock (&__pthread_key_lock);
   return 0;
 }
+strong_alias (__pthread_key_create, pthread_key_create)
+hidden_def (__pthread_key_create)
